activerecord/lib/active_record/encryption/extended_deterministic_queries.rb

```diff
@@ -50,7 +50,20 @@ def process_arguments(owner, args, check_for_additional_values)
               owner.deterministic_encrypted_attributes&.each do |attribute_name|
                 type = owner.type_for_attribute(attribute_name)
                 if !type.previous_types.empty? && value = options[attribute_name.to_s]
-                  options[attribute_name] = process_encrypted_query_argument(value, check_for_additional_values, type)
+                  options[attribute_name.to_s] = process_encrypted_query_argument(value, check_for_additional_values, type)
```

Comment:
This fixed an issue in introduced in #46809, where the set option could be a symbol instead of a string, resulting in a duplicated key (one symbol, one string). Let me know if you'd like me to pull this out into a separate PR.

Reply:
Pulled out into separate PR: #46813.